velox/functions/prestosql/window/Rank.cpp

```diff
@@ -51,22 +67,42 @@ class RankFunction : public exec::WindowFunction {
         previousPeerCount_ = 0;
       }
 
-      rawValues[resultOffset + i] = rank_;
-      previousPeerCount_ += 1;
+      switch (type) {
+        case velox::exec::RankType::kRank: {
+          ((int64_t*)rawValues)[resultOffset + i] = rank_;
+          previousPeerCount_ += 1;
+          break;
+        }
+        case velox::exec::RankType::kDenseRank: {
+          ((int64_t*)rawValues)[resultOffset + i] = rank_;
+          previousPeerCount_ = 1;
+          break;
+        }
+        case velox::exec::RankType::kPercentRank: {
+          ((double*)rawValues)[resultOffset + i] =
+              double(rank_ - 1) / (numPartitionRows_ - 1);
```

Comment:
If numPartitionRows_ is 1, then this would have a  divide by 0 exception. Is there any other way to calculate percentRank that avoids it ?

Reply:
Thanks for pointing it out, percent rank is now set to 0 for this case (consistent with PrestoDB).